scapy/contrib/isotp.py

```diff
@@ -433,8 +434,12 @@ def __init__(self,
         self.timeout = timeout
         self.filter_warning_emitted = False
 
-        def can_send(load):
-            can_socket.send(CAN(identifier=sid, data=load))
+        if padding:
+            def can_send(load):
+                can_socket.send(CAN(identifier=sid, data=load, length=8))
```

Comment:
Could you please modify the load here, instead of modifying the length field.

Reply:
Changed in:

https://github.com/secdev/scapy/pull/1595/commits/b4d56afed3cae1bd6807549b8633d6b86b32e0a0#diff-a11ae4af4a4fabd503da745c5589f9beR439